src/app.ts

```diff
@@ -2,15 +2,21 @@ import express from 'express';
 import bodyParser from 'body-parser';
 import { commonErrorHandler } from './lib/common-error-handler';
 
+import dbInit from './db/db';
+import {dataSourceRouter, dataSourcesPath} from './data-source.router';
+
+dbInit();
 const app = express();
 
 app.use(bodyParser.json());
+
+app.use(dataSourcesPath, dataSourceRouter);
+
 app.use(commonErrorHandler);
 
 app.listen(8000, err => {
-  console.log('server running');
   if (err) {
-    return console.log(err);
+    return console.error(err);
   }
 });
 
```

Comment:
Consider implementing an async function (e.g. run()), that will first instantiate mongoose and then call app.listen using async await syntax.